### PR TITLE
fix #2494 Clarify Mono.thenReturn javadoc

### LINKED ISSUE #2494: thenReturn documentation doesn't describe behavior for errors
## Documentation Issue
from reading the [documentation for Mono.thenReturn](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#thenReturn-V-), I got the impression that if the previous Mono completed with an error, it would be discarded and the return value would be returned instead. After experimenting I learned this is not the case, and errors in the preceding pipeline will skip the thenReturn operator and be propagated downstream.

## Improvement Suggestion

Clarify the error handling behavior of the thenReturn operator in the text of the documentation entry, using [the entry for Mono.then](https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Mono.html#then--) as an example. Modifying the marble diagram in a similar fashion would also be helpful, but is not critical for clarity of behavior.

### PR DESCRIPTION
**Pull request by bbukaty**

### CODE REVIEW

**Comment by bbukaty** [comment]
I'm not sure why this didn't seem to properly link to the issue #2494 .

**Comment by pivotal-issuemaster** [comment]
@bbukaty Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=2533)!

[Click here](https://cla.pivotal.io/sync/reactor?repositoryId=reactor/reactor-core&pullRequestId=2533) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@bbukaty Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/reactor?repositoryId=reactor/reactor-core&pullRequestId=2533)!

File: reactor-core/src/main/java/reactor/core/publisher/Mono.java [hunk 1]
```diff
@@ -4326,9 +4326,9 @@ public final void subscribe(Subscriber<? super T> actual) {
 	}
 
 	/**
-	 * Let this {@link Mono} complete then play another {@link Publisher}.
+	 * Let this {@link Mono} complete successfully then play another {@link Publisher}. On an error in the original {@link Mono}, the error signal is propagated instead.
 	 * <p>
-	 * In other words ignore element from this mono and transform the completion signal into a
+	 * In other words ignore elements from this mono and transform the completion signal into a
```
Comment (simonbasle) [inline reactor-core/src/main/java/reactor/core/publisher/Mono.java:4331]: the singular is on purpose here, as a `Mono` can only have one element.

**Review by simonbasle — CHANGES_REQUESTED** [review]
thanks for the PR @bbukaty ! one small nitpick on the change, otherwise looks good

**Review by simonbasle — APPROVED** [review]

**Pull request merged**